packages/desmos-std/src/profiles/mocks.rs

```diff
@@ -0,0 +1,408 @@
+        UserBlock {
+            blocker: Addr::unchecked("desmos1nwp8gxrnmrsrzjdhvk47vvmthzxjtphgxp5ftc"),
+            blocked: Addr::unchecked("desmos1rfv0f7mx7w9d3jv3h803u38vqym9ygg344asm3"),
+            reason: "test".to_string(),
+            subspace_id: "1".to_string(),
+        }
+    }
+
+    pub fn get_mock_chain_link() -> ChainLink {
+        ChainLink{
+            user: Addr::unchecked("desmos1nwp8gxrnmrsrzjdhvk47vvmthzxjtphgxp5ftc"),
+            address: ChainLinkAddr {
+                proto_type: "/desmos.profiles.v1beta1.Bech32Address".to_string(),
+                value: "cosmos18xnmlzqrqr6zt526pnczxe65zk3f4xgmndpxn2".to_string(),
+                prefix: "cosmos".to_string()
+            },
+            proof: Proof {
+                pub_key: PubKey {
+                    proto_type: "/cosmos.crypto.secp256k1.PubKey".to_string(),
+                    key: "AyRUhKXAY6zOCjjFkPN78Q29sBKHjUx4VSZQ4HXh66IM".to_string()
+                },
+                signature: Signature {
+                    proto_type: "/desmos.profiles.v1beta1.SingleSignatureData".to_string(),
+                    signature: "C7xppu4C4S3dgeC9TVqhyGN1hbMnMbnmWgXQI2WE8t0oHIHhDTqXyZgzhNNYiBO7ulno3G8EXO3Ep5KMFngyFg".to_string()
+                },
+                plain_text: "636f736d6f733138786e6d6c7a71727172367a74353236706e637a786536357a6b33663478676d6e6470786e32".to_string()
+            },
+            chain_config: ChainConfig { name: "cosmos".to_string() },
+            creation_time: "2022-02-21T13:18:57.800827Z".to_string()
+        }
+    }
+
+    pub fn get_mock_application_link() -> ApplicationLink {
+        ApplicationLink {
+            user: Addr::unchecked("desmos1nwp8gxrnmrsrzjdhvk47vvmthzxjtphgxp5ftc"),
+            data: Data {
+                application: "twitter".to_string(),
+                username: "goldrake".to_string(),
+            },
+            state: "APPLICATION_LINK_STATE_VERIFICATION_SUCCESS".to_string(),
+            oracle_request: OracleRequest {
+                id: "537807".to_string(),
+                oracle_script_id: "32".to_string(),
+                call_data: CallData {
+                    application: "twitter".to_string(),
+                    call_data: "7b22757365726e616d65223a224c756361675f5f2335323337227d".to_string(),
+                },
+                client_id: "desmos1nwp8gxrnmrsrzjdhvk47vvmthzxjtphgxp5ftc-twitter-goldrake"
+                    .to_string(),
+            },
+            result: AppLinkResult::Success {
+                value: "4c756361675f5f2345423337".to_string(),
+                signature: "9690d734171298eb4cc9636c36d8507535264c1fdb136c9095a6a50c41ccffa"
+                    .to_string(),
+            },
+            creation_time: "2022-02-21T13:18:57.800827Z".to_string(),
+        }
+    }
+}
+
+/// Replacement for cosmwasm_std::testing::mock_dependencies
+/// this use our CustomQuerier to use desmos querier
+pub fn mock_dependencies_with_custom_querier(
```

Comment:
I am trying about if we can mock all the modules' `QueryRouter` to one `deps`.
Currently it seems impossible since the generic T of `MockQuerier<T>` does not accept trait as parameter, so the `deps` only can be mocked for each modules individually, as a result, the following code cannot be tested:
```rust
fn query(deps: Deps) -> StdResult<QueryResponse> {
    let subspaces_querier = SubspacesQuerier::new(deps.querier.deref());
    let subspace = subspaces_querier.query_subspace(1)
    let profiles_querier = ProfilesQuerier::new(deps.querier.deref());
    let profile = profiles_querier.query_profile("user");
    QueryResponse{
       subspace, profile
   }
}
```
Should we have a general `DesmosQueryRouter` to solve the problem? Or just require developer only use one querier in one function to let the function be testable.
Any idea of it @bragaz ?

Reply:
I guess it would be better to only have one generic `DesmosQueryRouter` with a generic `QueryRoute` enum that contains all the nested desmos queries like we did at first:

```rust
#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
#[serde(rename_all = "snake_case")]
pub struct DesmosQueryRouter {
    pub route: DesmosRoute,
    pub query_data: QueryRoute ,
}

pub enum QueryRoute {
    Profiles(ProfilesQueryRoute)
    Subspaces(SubspacesQueryRoute)
}
```

We shouldn't limit the possibility to make more than one custom query per function IMO.

Reply:
@bragaz I agree with the idea. It is also easy to be testable as well.